osf/utils/migrations.py

```diff
@@ -0,0 +1,88 @@
+import os
+import json
+import logging
+
+from website import settings
+from osf.models import NodeLicense, MetaSchema
+from website.project.metadata.schemas import OSF_META_SCHEMAS
+
+logger = logging.getLogger(__file__)
+
+
+def ensure_licenses(*args, **kwargs):
+    """Upsert the licenses in our database based on a JSON file.
+
+    :return tuple: (number inserted, number updated)
+
+    Moved from website/project/licenses/__init__.py
+    """
+    ninserted = 0
+    nupdated = 0
+    with open(
+            os.path.join(
+                settings.APP_PATH,
+                'node_modules', 'list-of-licenses', 'dist', 'list-of-licenses.json'
+            )
+    ) as fp:
+        licenses = json.loads(fp.read())
+        for id, info in licenses.items():
+            name = info['name']
+            text = info['text']
+            properties = info.get('properties', [])
+
+            node_license, created = NodeLicense.objects.get_or_create(license_id=id)
+
+            node_license.name = name
+            node_license.text = text
+            node_license.properties = properties
+            node_license.save()
+
+            if created:
+                ninserted += 1
+            else:
+                nupdated += 1
+
+            logger.info('License {name} ({id}) added to the database.'.format(name=name, id=id))
+
+    logger.info('{} licenses inserted into the database, {} licenses updated in the database.'.format(
+        ninserted, nupdated
+    ))
+
+    return ninserted, nupdated
+
+
+def remove_licenses(*args):
+    pre_count = NodeLicense.objects.all().count()
+    NodeLicense.objects.all().delete()
+
+    logger.info('{} licenses removed from the database.'.format(pre_count))
+
+
+def ensure_schemas(*args):
```

Comment:
This differs slightly from the implementation on the latest develop, specifically in the use of `get_or_create` vs. `update_or_create`.

On `develop`:

```python
def ensure_schema(schema, name, version=1, active=True):
    MetaSchema = apps.get_model('osf.MetaSchema')
    MetaSchema.objects.update_or_create(
        name=name,
        schema_version=version,
        defaults={
            'schema': schema,
            'active': active
        }
    )

def ensure_schemas():
    """Import meta-data schemas from JSON to database if not already loaded
    """
    for schema in OSF_META_SCHEMAS:
        ensure_schema(schema, schema['name'], version=schema.get('version', 1), active=schema.get('active', True))
```

Is there a reason to change this?